packages/components/tree-select/src/tree-select.vue

```diff
@@ -0,0 +1,76 @@
+<script lang="ts">
+import { defineComponent, ref, h, reactive, computed, onMounted } from 'vue'
+import { pick } from 'lodash-unified'
+import ElSelect from '@element-plus/components/select'
+import ElTree from '@element-plus/components/tree'
+import { useSelect } from './select'
+import { useTree } from './tree'
+
+export default defineComponent({
+  name: 'ElTreeSelect',
+  props: {
+    ...ElSelect.props,
+    ...ElTree.props,
+  },
+  setup(props, context) {
+    const { slots, expose } = context
+
+    const select = ref<InstanceType<typeof ElSelect>>()
+    const tree = ref<InstanceType<typeof ElTree>>()
+
+    const key = computed(() => props.valueKey || props.nodeKey || 'value')
+
+    const selectProps = useSelect(props, context, { select, tree, key })
+    const treeProps = useTree(props, context, { select, tree, key })
+
+    // expose ElTree/ElSelect methods
+    const methods = reactive({})
+    expose(methods)
+    onMounted(() => {
+      Object.assign(methods, {
+        ...pick(tree.value, [
+          'filter',
+          'updateKeyChildren',
+          'getCheckedNodes',
+          'setCheckedNodes',
+          'getCheckedKeys',
+          'setCheckedKeys',
+          'setChecked',
+          'getHalfCheckedNodes',
+          'getHalfCheckedKeys',
+          'getCurrentKey',
+          'getCurrentNode',
+          'setCurrentKey',
+          'setCurrentNode',
+          'getNode',
+          'remove',
+          'append',
+          'insertBefore',
+          'insertAfter',
+        ]),
+        ...pick(select.value, ['focus', 'blur']),
+      })
+    })
+
+    return () =>
+      h(
+        ElSelect,
+        reactive({
+          ...selectProps,
+          ref: (ref) => (select.value = ref),
+        }),
```

Comment:
**Question:** was this really necessary? I'm not sure, would you elaborate a little bit about this?

Reply:
https://github.com/yujinpan/element-plus/commit/6d148dad3a11f89c3731a92124c3c9f41ae58276

1. The `props` is processed into `Refs`, but `v-bind` and render function props cannot read `Refs`, so use `reactive` unwrap the `Refs` and keep reactive.
2. The keyword `ref` requires `Ref`, but `reactive` broke it, so use function.